src/extractor/edge_based_graph_factory.cpp

```diff
@@ -172,6 +173,8 @@ void EdgeBasedGraphFactory::InsertEdgeBasedNode(const NodeID node_u, const NodeI
     }
 
     BOOST_ASSERT(current_edge_source_coordinate_id == node_v);
+
+    return Mapping{node_u, node_v, forward_data.edge_id, reverse_data.edge_id};
```

Comment:
see below. I feel this could become kind of messy. Its a mapping tailored to MLD but integrated into our general data structures.